server/src/test/java/com/broadcom/lsp/cobol/positive/FolderTextRegistry.java

```diff
@@ -0,0 +1,71 @@
+package com.broadcom.lsp.cobol.positive;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.ListMultimap;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.List;
+
+/** Registry for Cobol source code files. */
+@Slf4j
+public class FolderTextRegistry implements CobolTextRegistry {
+  private static final String POSITIVE_ENTRY = "positive";
+  private static final String NEGATIVE_ENTRY = "negative";
+  private static final String COPYBOOK_ENTRY = "copybooks";
+
+  private final ListMultimap<String, CobolText> texts = ArrayListMultimap.create();
+
+  public FolderTextRegistry(String pathToTestResources) {
+    if (StringUtils.isNotEmpty(pathToTestResources)) {
+        try {
+            collectFilesToTest(pathToTestResources);
+        } catch (IOException e) {
+            LOG.error(e.getMessage());
+        }
+    }
+  }
+
+  @Override
+  public List<CobolText> getPositives() {
+    return Collections.unmodifiableList(texts.get(POSITIVE_ENTRY));
+  }
+
+  @Override
+  public List<CobolText> getNegatives() {
+        return Collections.unmodifiableList(texts.get(NEGATIVE_ENTRY));
+    }
+
+  @Override
+  public List<CobolText> getCopybooks() {
+    return Collections.unmodifiableList(texts.get(COPYBOOK_ENTRY));
+  }
+
+  private void collectFilesToTest(String folderPath) throws IOException {
+    Files.walk(Paths.get(folderPath))
+      .filter(Files::isRegularFile)
+      .forEach(this::processFile);
+  }
+
+  private void processFile(Path path) {
+    String[] parts = path.toString().split("/");
+    if (parts.length < 2) {
```

Comment:
Why do we need this restriction?
It matches `folder/file.ext`, and also matches `file.ext`. But in the last case we will have an error in line:
```
String folder = parts[parts.length - 2];
```